### PR TITLE
Make finality verifier pallet instantiable

### LINKED ISSUE #360: Make Substrate Pallet Instantiable
We want a runtime to be able to bridge to many different Substrate chains. To do that we'll need to make it [instantiable](https://substrate.dev/recipes/instantiable.html). We may also need to add items to the configuration trait to account for differences in Substrate chains, such as different number, hash, and header types.

### PR DESCRIPTION
**Pull request by svyatonik**
closes #360 

I needed that for #824

### CODE REVIEW

File: modules/finality-verifier/src/lib.rs [hunk 1]
```diff
@@ -70,7 +70,7 @@ pub mod pallet {
 	use frame_system::pallet_prelude::*;
 
 	#[pallet::config]
-	pub trait Config: frame_system::Config {
+	pub trait Config<I: 'static = ()>: frame_system::Config {
```
Comment (MaciejBaj) [inline modules/finality-verifier/src/lib.rs:73]: is `I: 'static = ()` equal `I: DefaultInstance`?

Comment (svyatonik) [reply modules/finality-verifier/src/lib.rs:73]: I guess so :) From what I've learned from this new pallet declaration macros docs, now 'instance' should only be `'static`. Previously, with `decl_module! {}`, it had to implement `Instance` trait. I've been copypasting this from the `balances` pallet, which atm is the only pallet that uses this new pallet declaration syntax and supports multiple instances. There, `()` is used. Tbh, I'm not even sure if `DefaultInstance` is still being generated by the `#[pallet]` :)

Comment (MaciejBaj) [reply modules/finality-verifier/src/lib.rs:73]: cool, thanks a lot!

**Review by tomusdrw — APPROVED** [review]

File: modules/finality-verifier/src/lib.rs (hunk 1 above)
Comment (HCastano) [reply modules/finality-verifier/src/lib.rs:73]: So according to Gui `DefaultInstance` is not being supported anymore, and `()` should be used as the "default".

**Review by HCastano — APPROVED** [review]

**Pull request merged**